app/controllers/samples_controller.rb

```diff
@@ -967,15 +967,18 @@ def results_folder
     end
   end
 
+  # GET /samples/:id/stage_results
   def stage_results
-    @results = {}
-    @sample.first_pipeline_run.pipeline_run_stages.each do |stage|
-      if stage.name == "Experimental" && !current_user.admin?
-        next
+    if current_user.allowed_feature_list.include?("pipeline_viz")
+      @results = {}
+      @sample.first_pipeline_run.pipeline_run_stages.each do |stage|
+        if stage.name == "Experimental" && !current_user.admin?
+          next
```

Comment:
You can avoid `next` (not great) by changing the condition a bit:
`if stage.name != "Experimental" || current_user.admin?`